fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -2476,20 +2477,23 @@ class DDTeamCollectionImpl {
 					}
 				}
 				numSSPerAddr.clear();
+				numSSIgnoredPerAddr.clear();
 				hasHealthyTeam = (self->healthyTeamCount != 0);
 				RecruitStorageRequest rsr;
 				std::set<AddressExclusion> exclusions;
 				// Exclude existing servers running SS from being recruited again.
 				for (auto s = self->server_and_tss_info.begin(); s != self->server_and_tss_info.end(); ++s) {
 					auto serverStatus = self->server_status.get(s->second->getLastKnownInterface().id());
+					auto addr = s->second->getLastKnownInterface().stableAddress();
+					AddressExclusion addrExcl(addr.ip, addr.port);
 					if (serverStatus.excludeOnRecruit()) {
 						TraceEvent(SevDebug, "DDRecruitExcl1")
 						    .detail("Primary", self->primary)
 						    .detail("Excluding", s->second->getLastKnownInterface().address());
-						auto addr = s->second->getLastKnownInterface().stableAddress();
-						AddressExclusion addrExcl(addr.ip, addr.port);
 						exclusions.insert(addrExcl);
 						numSSPerAddr[addrExcl]++; // increase from 0
+					} else {
+						numSSIgnoredPerAddr[addrExcl]++;
 					}
```

Comment:
Looking at the implementation of `numExistingSSOnAddr`, may be we can just move `numSSPerAddr` out of the if block to make it does the same thing as `numExistingSSOnAddr`.
Current fix may still have a problem when there is a worker with 1 good SS and a failed SS.

Reply:
Well the one good SS will already mark this host as excluded with DDExcl1 in the code block above

Reply:
Yeah, I mean we can add both good SS and bad SS to `numSSPerAddr` because in `numExistingSSOnAddr`, which is used at the beginning of `initializeStorage` count both good SS and bad SS.